xlive/Blam/Engine/networking/NetworkMessageTypeCollection.cpp

```diff
@@ -283,20 +283,29 @@ void __stdcall handle_channel_message_hook(void* thisx, int network_channel_inde
 	if (!MessageIsCustom(message_type))
 		p_handle_channel_message(thisx, network_channel_index, message_type, dynamic_data_size, packet);
 
-	if (message_type == _player_add) {
+	switch (message_type)
+	{
+	case _membership_update:
+		network_session_membership_update_local_players_teams();
+		break;
+	case _player_add:
 		if (peer_network_channel->is_channel_state_5()
 			&& peer_network_channel->get_network_address(&addr))
 		{
-			auto peer_index = NetworkSession::GetPeerIndexFromNetworkAddress(&addr);
+			int32 peer_index = NetworkSession::GetPeerIndexFromNetworkAddress(&addr);
 			EventHandler::NetworkPlayerEventExecute(EventExecutionType::execute_after, peer_index, EventHandler::NetworkPlayerEventType::add);
 			NetworkMessage::SendAntiCheat(peer_index);
 		}
+		break;
+	case k_network_message_type_collection_count:
+	default:
```

Comment:
remove the count case and add:
DISPLAY_ASSERT("unreachable")

Reply:
no, because it shouldn't throw an error. Going to remove the unnecessary cases

Reply:
ah my bad, realized it's a hook and not the actual switch case.